swap_meet/vendor.py

```diff
@@ -1,2 +1,71 @@
+from swap_meet.item import Item
+#wave1
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+    # inventory is an empty list
+        if inventory is None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        self.item = item           # item (this method takes only 1 item)
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item): 
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+# wave 2
+    def get_by_category(self, category):
+        items_inventory = []
+        for item in self.inventory:
+            if item.category == category:
+                items_inventory.append(item)
+        return items_inventory
+        
+# wave 3
+    def swap_items(self, friend_vendor, my_item, their_item): 
+
+        if my_item in self.inventory and their_item in friend_vendor.inventory:
+            self.remove(my_item)
+            self.add(their_item)
+            friend_vendor.inventory.append(my_item)
+
+            friend_vendor.remove(their_item)
```

Comment:
Great use of helper methods!  I recommend using them whenever possible:
```suggestion
            self.remove(my_item)
            self.add(their_item)
            friend_vendor.add(my_item)

            friend_vendor.remove(their_item)
```